common/persistence/cassandra/cassandraVisibilityPersistence.go

```diff
@@ -44,7 +44,8 @@ const (
 	defaultCloseTTLSeconds = 86400
 	openExecutionTTLBuffer = int64(86400) // setting it to a day to account for shard going down
 
-	maxCassandraTTL = int64(630720000) // Cassandra TTL maximum, 20 years in second
+	// ref: https://docs.datastax.com/en/dse-trblshoot/doc/troubleshooting/recoveringTtlYear2038Problem.html
+	maxCassandraTTL = int64(315360000) // Cassandra max support time is 2038-01-19T03:14:06+00:00. Updated this to 10 years to support until year 2028
```

Comment:
Why not
```
var maxCassandraTTL = (time.Date(2038,01,19,03,14,06,0,time.UTC).Sub(time.Now().UTC())).Milliseconds()
```

Reply:
We have default timeouts of 10 years.

Reply:
depending on whether we are still going to support Cassandra after 10 years
the solution from uber is much more simpler.

@samarabbas what do you think?

Reply:
Agreed with @wxing1292.  The main issue is we need a concept of infinite timeout in Temporal.  Using a default of 10 years introduces a scalability bottleneck.  Here are couple of issues tracking this:
https://github.com/temporalio/temporal/issues/358
https://github.com/temporalio/temporal/issues/573

Reply:
This is not even 10 but 8 years. I don't understand why they picked up some random number instead of properly set to the real limit Cassandra gives us. It is used in 2 places only.

Reply:
well, I guess as long as the time is really long (far in the future), then it is ok? BTW, cadence picked 5 years...